src/goTest.ts

```diff
@@ -32,8 +32,12 @@ export function testAtCursor(goConfig: vscode.WorkspaceConfiguration, args: any)
 		return;
 	}
 	if (editor.document.isDirty) {
```

Comment:
A little history...

The `run/debug test` codelens and the `Go: Test function at cursor` command required us to parse the file, find the name of the test method to pass the same to `go test`. This is why I had added the check on whether the file is dirty or not in the first place

Testing the package/workspace doesn't need us to do this

On the other hand testing current file also required us to parse the file and find the test methods.
So we should add the same "save" operation in the `testCurrentFile` method in this file

Reply:
Thanks for the background! 

Is it a correct understanding that all four methods then should still perform the save operation and now that we're performing it in `testAtCursor` the check for `editor.document.isDirty` is also no longer required?

Reply:
Which 4 methods? Cursor, File, Package and Workspace?

Cursor and File should have the file save. Package and Workspace dont, as you can run those even without any file open.

Yes, you can skip the check on dirty file and perform the save directly for cursor and file.